org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/part/DependencyEditorViewerPart.java

```diff
@@ -0,0 +1,267 @@
+			cellEditor.setItems(DEFAULT_TYPES);
+			return cellEditor;
+		};
+
+		return newEditingSupport(MavenTargetDependency::getType, MavenTargetDependency::setType, editorFactory);
+	}
+
+	private EditingSupport newEditingSupport(Function<MavenTargetDependency, String> getter,
+			BiConsumer<MavenTargetDependency, String> setter, Function<Composite, CellEditor> editorFactory) {
+		return new EditingSupport(viewer) {
+			@Override
+			protected CellEditor getCellEditor(Object element) {
+				return editorFactory.apply((Composite) getViewer().getControl());
+			}
+
+			@Override
+			protected boolean canEdit(Object element) {
+				return element instanceof MavenTargetDependency;
+			}
+
+			@Override
+			protected String getValue(Object element) {
+				return getter.apply((MavenTargetDependency) element);
+			}
+
+			@Override
+			protected void setValue(Object element, Object value) {
+				MavenTargetDependency dependency = (MavenTargetDependency) element;
+				String oldValue = getter.apply(dependency);
+				String newValue = (String) value;
+
+				// The table column needs to be redrawn. Otherwise we still see the old value...
+				BiConsumer<MavenTargetDependency, String> realSetter = (k, v) -> {
+					setter.accept(k, v);
+					getViewer().update(element, null);
+				};
+
+				model.modify(dependency, realSetter, oldValue, newValue);
+			}
+		};
+	}
+
+	/**
+	 * <p>
+	 * Creates a new column-specific comparator, wrapped around a selection adapter.
+	 * Whenever the column header is clicked, the table is sorted lexicographically
+	 * by this column.
+	 * </p>
+	 * <p>
+	 * If the viewer is sorted by a different column or if this column is clicked
+	 * for the first time, the elements are sorted in ascending order. If the column
+	 * is clicked again, elements are sorted in descending order.
+	 * </p>
+	 * 
+	 * @param getter The accessor for the property shown in the current column.
+	 * @return A selection adapter for the column header.
+	 */
+	private SelectionAdapter newSortAdapter(Function<MavenTargetDependency, String> getter) {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				Comparator<MavenTargetDependency> comparator;
+
+				// When we don't currently sort by this column, default to ascending order
+				if (viewer.getTable().getSortColumn() != e.widget || viewer.getTable().getSortDirection() == SWT.UP) {
+					viewer.getTable().setSortDirection(SWT.DOWN);
+					comparator = Comparator.comparing(getter);
+				} else {
+					viewer.getTable().setSortDirection(SWT.UP);
+					comparator = Comparator.comparing(getter).reversed();
+				}
+
+				viewer.getTable().setSortColumn((TableColumn) e.widget);
+				viewer.setComparator(new ViewerComparator() {
+					@Override
+					public int compare(Viewer viewer, Object e1, Object e2) {
+						return comparator.compare((MavenTargetDependency) e1, (MavenTargetDependency) e2);
+					}
+				});
+				viewer.refresh();
+			}
+		};
+	}
```

Comment:
Instead of a anonymous class you could simply use `SelectionListener.widgetSelectedAdapter()`
```suggestion
	private SelectionListener newSortAdapter(Function<MavenTargetDependency, String> getter) {
		return SelectionListener.widgetSelectedAdapter(e -> {
			Comparator<MavenTargetDependency> comparator;

			// When we don't currently sort by this column, default to ascending order
			if (viewer.getTable().getSortColumn() != e.widget || viewer.getTable().getSortDirection() == SWT.UP) {
				viewer.getTable().setSortDirection(SWT.DOWN);
				comparator = Comparator.comparing(getter);
			} else {
				viewer.getTable().setSortDirection(SWT.UP);
				comparator = Comparator.comparing(getter).reversed();
			}

			viewer.getTable().setSortColumn((TableColumn) e.widget);
			viewer.setComparator(new ViewerComparator() {
				@Override
				public int compare(Viewer viewer, Object e1, Object e2) {
					return comparator.compare((MavenTargetDependency) e1, (MavenTargetDependency) e2);
				}
			});
			viewer.refresh();
		});
	}
```